spanner/value.go

```diff
@@ -877,6 +877,96 @@ func (n *PGNumeric) UnmarshalJSON(payload []byte) error {
 	return nil
 }
 
+// NullProto represents a Cloud Spanner PROTO that may be NULL.
+type NullProto struct {
+	ProtoVal proto.Message // ProtoVal contains the value when it is non-NULL, and nil when NULL.
+	Valid    bool          // Valid is true if Proto is not NULL.
+}
+
+// IsNull implements NullableValue.IsNull for NullProto.
+func (n NullProto) IsNull() bool {
+	return !n.Valid
+}
+
+// String implements Stringer.String for NullProto.
+func (n NullProto) String() string {
+	if !n.Valid {
+		return nullString
+	}
+	return n.ProtoVal.String()
+}
+
+// MarshalJSON implements json.Marshaler.MarshalJSON for NullProto.
+func (n NullProto) MarshalJSON() ([]byte, error) {
+	if n.Valid {
+		return proto.Marshal(n.ProtoVal)
+	}
+	return jsonNullBytes, nil
+}
+
+// UnmarshalJSON implements json.Unmarshaler.UnmarshalJSON for NullProto.
+func (n *NullProto) UnmarshalJSON(payload []byte) error {
+	if payload == nil {
+		return fmt.Errorf("payload should not be nil")
+	}
+	if bytes.Equal(payload, jsonNullBytes) {
+		n.Valid = false
+		return nil
+	}
+	err := proto.Unmarshal(payload, n.ProtoVal)
+	if err != nil {
+		return fmt.Errorf("payload cannot be converted to a proto message: got %v, err: %s", string(payload), err)
+	}
+	n.Valid = true
+	return nil
+}
+
+// NullEnum represents a Cloud Spanner ENUM that may be NULL.
+type NullEnum struct {
+	EnumVal protoreflect.Enum // EnumVal contains the value when it is non-NULL, and nil when NULL.
```

Comment:
Same here: Is it `nil` or `protoreflect.Enum{}` when the value is NULL?

Reply:
`protoreflect.Enum` is an interface. The zero value or default value of an interface is `nil`.